client/src/toplevels/ViewHandler.res

```diff
@@ -34,20 +36,27 @@ let handlerIsExeFail = (vp: viewProps): bool =>
     |> Option.andThen(~f=Analysis.getLiveValue'(vp.analysisStore))
     |> Option.map(~f=outermostResult =>
       switch outermostResult {
-      | DIncomplete(_) | DError(_) | DErrorRail(_) => true
+      | RT.Dval.DIncomplete(_) | DError(_) | DErrorRail(_) => true
       | _ => false
       }
     )
     |> Option.unwrap(~default=false)
   }
 
 let triggerHandlerButton = (vp: viewProps, spec: PT.Handler.Spec.t): Html.html<msg> =>
-  switch (spec.space, spec.name, spec.modifier) {
+  switch spec {
   /* Hide button if spec is not filled out because trace id
    is needed to recover handler traces on refresh. */
-  | (F(_, a), F(_, b), F(_, c))
-    if List.any(~f=s => String.length(s) == 0, list{a, b, c}) => Vdom.noNode
-  | (F(_), F(_), F(_)) =>
```

Comment:
I'll need some h;elp understanding the change here. (Maybe I'll get context by looking at the specific relevant commit)